### PR TITLE
feat(dev): Remove implicit jest watch mode when running `yarn test`

### PR DESCRIPTION
**Pull request by billyvg**
A few reasons for this change:

* The behavior of `yarn test` was a bit unclear, there were a few conditions where it would not run in "watch" mode (CI, pre-commit, exporting coverage stats)
* We were having issues in "watch" mode due to `jest-sentry-environment`. Our wrapper script called `jest` via node API w/ "watch" mode as an arg, while `jest-sentry-environment` checked the global `process.argv` for "watch" mode. This is because the run options are not passed to the jest environment, and there is no other way to determine "watch" mode. (Note: this is fixed in v28 of jest, but it is currently in pre-release: https://github.com/facebook/jest/issues/12271).

Refactor this a bit so that `yarn test` calls `test-base` with `--watch` in cli args and we remove this depending on env vars. This is to keep compatibility with existing workflow of running `yarn test` and expecting watch mode. Explicitly call `test-base` when testing in CI.

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]
big change.

Let's send an email?

**Comment by billyvg** [comment]
@evanpurkhiser thoughts on inverting this a bit and making `yarn test` developer-focused (i.e. it runs in watch mode), and add a `test-base` or something?

**Comment by evanpurkhiser** [comment]
I'm down to invert it. Then we don't have to re-train our muscle memory

File: package.json [hunk 1]
```diff
@@ -206,11 +206,12 @@
   "APIMethod": "stub",
   "proxyURL": "http://localhost:8000",
   "scripts": {
-    "test": "node scripts/test.js",
-    "test-precommit": "yarn test --bail --findRelatedTests -u",
-    "test-ci": "yarn test --ci --coverage --runInBand",
```
Comment (Zylphrex) [inline package.json:211]: Is `--runInBand` being removed intentionally?

File: package.json [hunk 2]
```diff
@@ -206,11 +206,12 @@
...
+    "test-wrap": "node scripts/test.js",
+    "test": "yarn test-wrap --watch",
+    "test-ci": "yarn test-wrap --ci --coverage",
```
Comment (billyvg) [inline package.json:211]:
```suggestion
    "test-ci": "yarn test-wrap --ci --coverage --runInBand",
```

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**